src/Umbraco.Core/Auditing/IdentityAuditEventArgs.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Threading;
+using System.Web;
+using Umbraco.Core.Security;
+
+namespace Umbraco.Core.Auditing
+{
+    /// <summary>
+    /// This class is used by events raised from hthe BackofficeUserManager
+    /// </summary>
+    public class IdentityAuditEventArgs : EventArgs
+    {
+        /// <summary>
+        /// The action that got triggered from the audit event
+        /// </summary>
+        public AuditEvent Action { get; set; }
+
+        /// <summary>
+        /// Current date/time in UTC format
+        /// </summary>
+        public DateTime DateTimeUtc { get; private set; }
+
+        /// <summary>
+        /// The source IP address of the user performing the action
+        /// </summary>
+        public string IpAddress { get; set; }
+
+        /// <summary>
+        /// The user affected by the event raised
+        /// </summary>
+        public int AffectedUser { get; set; }
+
+        /// <summary>
+        /// If a user is perfoming an action on a different user, then this will be set. Otherwise it will be -1
+        /// </summary>
+        public int PerformingUser { get; set; }
+
+        /// <summary>
+        /// An optional comment about the action being logged
+        /// </summary>
+        public string Comment { get; set; }
+
+        /// <summary>
+        /// This property is always empty except in the LoginFailed event for an unknown user trying to login
+        /// </summary>
+        public string Username { get; set; }
+
+        /// <summary>
+        /// Sets the properties on the event being raised, all parameters are optional except for the action being performed
+        /// </summary>
+        /// <param name="action">An action based on the AuditEvent enum</param>
+        /// <param name="ipAddress">The client's IP address. This is usually automatically set but could be overridden if necessary</param>
+        /// <param name="performingUser">The Id of the user performing the action (if different from the user affected by the action)</param>
+        public IdentityAuditEventArgs(AuditEvent action, string ipAddress = "", int performingUser = -1)
+        {
+            DateTimeUtc = DateTime.UtcNow;
+            Action = action;
+
+            IpAddress = string.IsNullOrWhiteSpace(ipAddress)
+                ? GetCurrentRequestIpAddress()
+                : ipAddress;
+
+            PerformingUser = performingUser == -1
+                ? GetCurrentRequestBackofficeUserId()
+                : performingUser;
+        }
+
+        /// <summary>
+        /// Returns the current request IP address for logging if there is one
+        /// </summary>
+        /// <returns></returns>
+        protected string GetCurrentRequestIpAddress()
+        {
+            var httpContext = HttpContext.Current == null ? (HttpContextBase)null : new HttpContextWrapper(HttpContext.Current);
```

Comment:
The IP address should really be set via the ctor of the class not by using Singletons

Reply:
I've pushed a fix for this